### ISSUE TITLE
Checkbox is not accessible

### ISSUE THREAD

**Issue by charlag** [tutao/tutanota#4607]
The label is not associated with the checkbox. Since we can't really use static IDs we should instead nest checkbox inside the label.

# Test Notes
- [ ] Screen readers announce the checkbox identically to a standard checkbox element
- [ ] Clicking on the checkbox's box toggles the state
- [ ] Clicking on the checkbox's label toggles the state
- [ ] The checkbox is still positioned correctly

**Comment by murilopereirame** [comment]
The checkbox is misplaced in iOS and Android. 
Double tapping the label works, but the user can't select the checkbox itself so easily, as it is behind the text (Android), that makes the screen reader not announce the state.

iOS
![IMG_0074](https://github.com/tutao/tutanota/assets/34790144/11e07d5e-5275-4284-aa32-a1b989f7472e)

Android
![Screenshot_20231128-144947](https://github.com/tutao/tutanota/assets/34790144/8864e2ff-e203-4dd4-acdb-5c21461ee0f7)
